Tests/MongoSwiftSyncTests/UnifiedTestRunner/EntityDescription.swift

```diff
@@ -115,3 +117,155 @@ enum EntityDescription: Decodable {
+        self.ignoreEvents = (ignoreEvents ?? []) + ["configureFailPoint"]
+    }
+
+    public func handleCommandEvent(_ event: CommandEvent) {
+        guard self.monitoring else {
+            return
+        }
+        guard self.observeEvents.contains(event.type) else {
+            return
+        }
+        guard !self.ignoreEvents.contains(event.commandName) else {
+            return
+        }
+        self.events.append(event)
+    }
+
+    func enable() throws {
+        guard !self.monitoring else {
+            throw TestError(message: "TestCommandMonitor is already enabled")
+        }
+        self.monitoring = true
+    }
+
+    func disable() throws -> [CommandEvent] {
+        guard self.monitoring else {
+            throw TestError(message: "TestCommandMonitor is already disabled")
+        }
+        self.monitoring = false
+        defer { self.events.removeAll() }
+        return self.events
+    }
+}
+
+/// Represents an entity created at the start of a test.
+enum Entity {
+    case client(UnifiedTestClient)
+    case database(MongoDatabase)
+    case collection(MongoCollection<BSONDocument>)
+    case session(ClientSession)
+
+    func asTestClient() throws -> UnifiedTestClient {
+        guard case let .client(client) = self else {
+            throw TestError(message: "Failed to return entity \(self) as a client")
+        }
+        return client
+    }
+
+    func asDatabase() throws -> MongoDatabase {
+        guard case let .database(db) = self else {
+            throw TestError(message: "Failed to return entity \(self) as a database")
+        }
+        return db
+    }
+
+    func asCollection() throws -> MongoCollection<BSONDocument> {
+        guard case let .collection(coll) = self else {
+            throw TestError(message: "Failed to return entity \(self) as a collection")
+        }
+        return coll
+    }
+
+    func asSession() throws -> ClientSession {
+        guard case let .session(session) = self else {
+            throw TestError(message: "Failed to return entity \(self) as a session")
+        }
+        return session
+    }
+}
+
+extension Array where Element == EntityDescription {
+    /// Converts an array of entity descriptions from a test file into an entity map.
+    func toEntityMap() throws -> [String: Entity] {
+        var map = [String: Entity]()
+        for desc in self {
+            switch desc {
+            case let .client(clientDesc):
+                map[clientDesc.id] = try .client(UnifiedTestClient(clientDesc))
+            case let .database(dbDesc):
+                guard let clientEntity = try map[dbDesc.client]?.asTestClient() else {
+                    throw TestError(message: "No client with id \(dbDesc.client) found in entity map")
+                }
+                map[dbDesc.id] = .database(clientEntity.client.db(
+                    dbDesc.databaseName,
+                    options: dbDesc.databaseOptions
+                )
+                )
```

Comment:
this double paren is a little weird, I guess SwiftFormat doesn't mind it though. Could we move them to one line like the similar case below?